src/apphosting/secrets/index.ts

```diff
@@ -164,3 +174,88 @@ export async function upsertSecret(
   // This may not be a great idea until version control is actually implemented.
   return false;
 }
+
+/**
+ * Fetches secrets from Google Secret Manager and returns their values in plain text.
+ */
+export async function fetchSecrets(
+  projectId: string,
+  secrets: Secret[],
+): Promise<Map<string, string>> {
+  let secretsKeyValuePairs: Map<string, string>;
+
+  try {
+    const secretPromises: Promise<[string, string]>[] = secrets.map(async (secretConfig) => {
+      const [name, version] = getSecretNameParts(secretConfig.secret!);
+
+      const value = await gcsm.accessSecretVersion(projectId, name, version);
+      return [secretConfig.variable, value] as [string, string];
+    });
+
+    const secretEntries = await Promise.all(secretPromises);
+    secretsKeyValuePairs = new Map(secretEntries);
+  } catch (e: any) {
+    throw new FirebaseError(`Error exporting secrets`, {
+      original: e,
+    });
+  }
+
+  return secretsKeyValuePairs;
+}
+
+/**
+ * Returns the appropriate App Hosting YAML configuration for exporting secrets.
+ *
+ * @returns The final merged config
+ */
+export async function loadConfigToExport(
+  cwd: string,
+  environmentConfigFile?: string,
+): Promise<AppHostingYamlConfig> {
+  if (environmentConfigFile && !APPHOSTING_YAML_FILE_REGEX.test(environmentConfigFile)) {
+    throw new FirebaseError(
+      "Invalid apphosting yaml config file provided. File must be in format: 'apphosting.yaml' or 'apphosting.<environment>.yaml'",
+    );
+  }
+
+  // Get all apphosting yaml files ignoring the apphosting.local.yaml file
+  const appHostingConfigPaths = discoverConfigsAtProjectRoot(cwd)?.filter(
+    (path) => !path.endsWith(APPHOSTING_LOCAL_YAML_FILE),
+  );
+  if (!appHostingConfigPaths) {
+    throw new FirebaseError("No apphosting configs found");
+  }
+
+  // generate a map to make it easier to interface between file name and it's path
+  const fileNameToPathMap: Map<string, string> = new Map();
+  for (const path of appHostingConfigPaths) {
+    const fileName = basename(path);
+    fileNameToPathMap.set(fileName, path);
+  }
+
+  const baseFilePath = fileNameToPathMap.get(APPHOSTING_BASE_YAML_FILE)!;
+  let environmentConfigFilePath = baseFilePath;
+  if (!environmentConfigFile) {
+    // If file is not provided, prompt the user
+    environmentConfigFilePath = await promptForAppHostingYaml(
+      fileNameToPathMap,
+      "Which environment would you like to export secrets from Secret Manager for?",
+    );
+  }
+
+  return await loadConfigForEnvironment(environmentConfigFilePath, baseFilePath);
```

Comment:
optional suggestion: This logic is still a little bit confusing to me. There's a lot of mixed conditionals and different handling of the base config depending on the inputs, along with aliasing vars (L237) that makes it a bit hard to understand what's going on.

This is very optional but it seems to me that there are two cases we need to handle here: 
1. User passed an explicit config file
2. User did not and we need to prompt.

My intuition says branching this function earlier may make this clearer, for example:

```
function(userGivenConfigFile) {
  allConfigs = discoverConfigs()

  // Get path for user-specified config file
  if (userGivenConfigFile) {
    userGivenConfigFilePath = allConfigs[userGivenConfigFile]
  } else {
    userGivenConfigFilePath = promptForConfig(allConfigs);   
  }
  
  // Load the config with any appropriate overrides
  if (userGivenConfigFile == BASE_YAML_FILE) {
    return AppHostingConfig.loadFile(BASE_YAML_FILE);
  }
  baseYamlFilePath = allConfigs[BASE_YAML_FILE];
  return loadConfigForEnvironment(userGivenConfigFile, baseYamlFilePath);
  }
}
```

Reply:
makes sense to me, this is definitely more clear!